onnxruntime/test/python/onnxruntime_test_python.py

```diff
@@ -121,6 +121,30 @@ def testModelSerializationWithExternalInitializers(self):  # noqa: N802
             else:
                 raise onnxruntime_error
 
+    def testModelSerializationWithExternalInitializersToDirectory(self):  # noqa: N802
+        try:
+            so = onnxrt.SessionOptions()
+            so.log_severity_level = 1
+            so.logid = "TestModelSerializationWithExternalInitializersToDirectory"
+            directory = "./testdata/"
+            so.optimized_model_filepath = os.path.join(directory, "model_with_external_initializers_in_dir.onnx")
+            external_initializers_file = "external_initializers_in_dir.bin"
+            so.add_session_config_entry(
+                "session.optimized_model_external_initializers_file_name", external_initializers_file
+            )
+            so.add_session_config_entry("session.optimized_model_external_initializers_min_size_in_bytes", "100")
+            onnxrt.InferenceSession(get_name("mnist.onnx"), sess_options=so, providers=["CPUExecutionProvider"])
+            self.assertTrue(os.path.isfile(so.optimized_model_filepath))
+            self.assertTrue(os.path.isfile(os.path.join(directory, external_initializers_file)))
+        except Fail as onnxruntime_error:
```

Comment:
Consider using with self.assertRaisesRegex? https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex

Reply:
This test case is written this way to follow the same format as the previous test case.

Reply:
Was there a reason it should be done the same way? It seems less idiomatic Python to prefer

Reply:
For any new code it seems preferable to follow best practices that are up to date so we don’t introduce new tech debt, and we can migrate the old code to do the same.

Reply:
I see the test is not really expecting the error. My mistake.